### DISCUSSION TITLE
QuarkusTest: How to inject object via constructor in Kotlin?

### DISCUSSION THREAD

**Discussion by softshipper** [quarkusio/quarkus#26484] [Q&A]
Hi all

I am using Kotlin to write Quarkus application and I have a question regarding to constructor.
I have a test class that looks as follows:

```
@QuarkusTest
class CertStoreGrpcTest {

    @GrpcClient("cert-desk-svc")
    lateinit var desk: CertStoreService

    @DisplayName("Create key- and truststore regarding user id")
    @Test
    fun testCreateUserCertStore() {
        println(desk)

        desk.createStore(
            CreateStoreRequest.newBuilder()
                .setStore(CertStore.newBuilder().setId("3b9553d6-2002-4485-9487-a660f23d553c")).build()
        )
    }

}
```

and would like to translate into 

```
class CertStoreGrpcTest(@GrpcClient("cert-desk-svc") val desk: CertStoreService) 
```

and the compiler complains:
```
When using constructor injection in a test, the only legal operation is to assign the constructor values to fields. Offending class is class ....

```

What does it mean and how to solve it?

Kind regards

[a comment by an automated account is not shown]

**Reply by evanchooly** [reply]
I've never seen that.  Can you paste in the entire stack trace with that message?

**Reply by softshipper** [reply]
Here we go:
[cert_desk_svc.zip](https://github.com/quarkusio/quarkus/files/9023030/cert_desk_svc.zip)
just run `./gradlew quarkusTest` in the console and  you will see the exceptions.
